vulture/core.py

```diff
@@ -42,13 +41,18 @@ def _is_special_name(name):
 
 
 def _match(name, patterns, case=True):
-    func = fnmatchcase if case else fnmatch
-    return any(func(name, pattern) for pattern in patterns)
+    if not case:
+        name = pathlib.Path(str(name).lower())
+    return any(name.match(pattern) for pattern in patterns)
+
+
+def _match_name(name, patterns):
+    return any(fnmatchcase(name, pattern) for pattern in patterns)
```

Comment:
I think it's better to let _match() only accept strings (callers must then convert to strings themselves) and remove the _match_name() function again.